controllers/operandconfig.go

```diff
@@ -142,7 +142,14 @@ func mergeChangedMap(key string, defaultMap interface{}, changedMap interface{},
 			if changedMap == nil {
 				finalMap[key] = defaultMap
 			} else {
-				finalMap[key], _ = rules.ResourceComparison(defaultMap, changedMap)
+				comparableKeys := []string{"replicas", "cpu", "memory"}
+				for _, comparable := range comparableKeys {
```

Comment:
not sure if we want to do the comparable key check like this. If we add more configurations in the future, and squads determine they can be changed after deployment, then comparing this way will not scale well.

Reply:
I think we can add the comparable key check since the ResourceComparison is for resource size comparison.

Reply:
right, but I mean in the future, if we add any more configurations, e.g. maybe operators include a new `healthcheck-interval` configuration that is the number of seconds between pinging a health check endpoint, and we want to add this to CS CR as well, then this new variable could be comparable (maybe we want to take the lowest value).

The way this code is checking for comparable keys won't scale very well if we add more configurations like this in the future. We would also need to keep updating the list every time. Should we consider this? Or is looping over a list of keys good enough?

Reply:
I think we keep them for now. The price to maintain the list is not high for us. 
Considering the case you list, we need to update the ResourceConparison function anyway without the comparable keys.